File: ui/web/public/types/Console.mjs

export const Console = function() {

	this.Messages = [];

};

Console.prototype = {

	GetMessages: function(from) {

		let result = [];

		if (this.Messages.length > 0 && from < this.Messages.length) {

			for (let m = from; m < this.Messages.length; m++) {
				result.push(this.Messages[m]);
			}

		}

		return result;

	}

};

File: source/ui/web/routes/session/Console.go
package session

import "exocomp/engine"
import "exocomp/ui/web/handlers"
import "encoding/json"
import "net/http"
import "strconv"

func Console(session *engine.Session, request *http.Request, response http.ResponseWriter) {

	if request.Method == http.MethodGet {

		if session.Console != nil {

			messages               := session.Console.GetMessages(0)
			response_payload, err0 := json.MarshalIndent(messages, "", "\t")

			if err0 == nil {

				response.Header().Set("Content-Type", "application/json")
				response.Header().Set("Content-Length", strconv.Itoa(len(response_payload)))
				response.WriteHeader(http.StatusOK)
				response.Write(response_payload)

			} else {
				handlers.InternalServerError(session, err0, request, response)
			}

		} else {
			handlers.NotFound(session, request, response)
		}

	} else {
		handlers.MethodNotAllowed(session, request, response)
	}

}
